service/src/main/java/uk/gov/hmcts/reform/fpl/service/document/ManageDocumentService.java

```diff
@@ -116,6 +137,197 @@ public class ManageDocumentService {
     private static final Predicate<Element<SupportingEvidenceBundle>> SOLICITOR_FILTER =
         bundle -> bundle.getValue().isUploadedByRepresentativeSolicitor();
 
+    private String getDocumentListActualFieldName(String fieldName) {
+        String[] splitFieldName = fieldName.split("\\.");
+        if (splitFieldName.length == 2) {
+            return splitFieldName[1];
+        } else {
+            return fieldName;
+        }
+    }
+
+    private Object getDocumentListHolder(String fieldName, CaseData caseData) throws Exception {
+        String[] splitFieldName = fieldName.split("\\.");
+        if (splitFieldName.length == 2) {
+            PropertyDescriptor pd = BeanUtils.getPropertyDescriptor(CaseData.class, splitFieldName[0]);
+            if (pd != null) {
+                return pd.getReadMethod().invoke(caseData);
+            } else {
+                throw new IllegalArgumentException("unable to get property descriptor from CaseData.class: "
+                    + splitFieldName[0]);
+            }
+        } else {
+            return caseData;
+        }
+    }
+
+    private String getDocumentListHolderFieldName(String fieldName) {
+        String[] splitFieldName = fieldName.split("\\.");
```

Comment:
this looks the same as getDocumentListActualFieldName?

Reply:
could maybe add all this splitFieldName logic into a single helper and reuse across these methods?